### PR TITLE
Update default fee percents

### PR DESCRIPTION
**Pull request by znarf**
- Host always with hostFeePercent and platformFeePercent set (if missing hostFeePercent=0, platformFeePercent=5)
- new Hosts will start with  hostFeePercent=0, platformFeePercent=0. 
- Collectives always starts with hostFeePercent=null, platformFeePercent=null
- When applying to an Host they inherit its hostFeePercent and platformFeePercent
- On Rejection or when they reset the Host, they're moved back to hostFeePercent=null, platformFeePercent=null
- Remove "covid-19" exception

### CODE REVIEW

File: migrations/20201020153732-update-default-feePercent.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      WHERE "HostCollectives"."id" = "Collectives"."HostCollectiveId"
+      AND "Collectives"."hostFeePercent" IS NULL
+    `);
+
+    // Update all unhosted collectives to hostFeePercent=null platformFeePercent=null
+    await queryInterface.sequelize.query(`
+      UPDATE "Collectives"
+      SET "platformFeePercent" = NULL, "hostFeePercent" = NULL
+      WHERE "HostCollectiveId" IS NULL
+      AND "type" IN ('COLLECTIVE', 'FUND', 'EVENT", "PROJECT')
```
Comment (kewitz) [inline migrations/20201020153732-update-default-feePercent.js:44]:
```suggestion
      AND "type" IN ('COLLECTIVE', 'FUND', 'EVENT', 'PROJECT')
```

File: server/graphql/v1/mutations/collectives.js [hunk 2]
```diff
@@ -466,13 +466,19 @@ export async function approveCollective(remoteUser, CollectiveId) {
...
     events.map(event => {
       event.update({ isActive: true, approvedAt: new Date() });
     }),
   );
+  const projects = await collective.getProjects();
+  await Promise.all(
+    projects.map(project => {
+      project.update({ isActive: true, approvedAt: new Date() });
+    }),
+  );
```
Comment (kewitz) [inline server/graphql/v1/mutations/collectives.js:481]: :+1:

File: server/graphql/v1/mutations/collectives.js [hunk 3]
```diff
@@ -518,10 +524,12 @@ export async function archiveCollective(_, args, req) {
     membership.destroy();
   }
 
-  if (collective.type === types.EVENT) {
+  if (collective.type === types.EVENT || collective.type === types.PROJECT) {
     return collective.update({ isActive: false, deactivatedAt: Date.now() });
   }
 
+  // TODO: cascade deactivation to EVENTs and PROJECTs?
+
```
Comment (kewitz) [inline server/graphql/v1/mutations/collectives.js:532]: I think we should cascade the deactivation, the exception should be not deactivating.

File: server/models/Collective.js [hunk 4]
```diff
@@ -702,18 +700,6 @@ export default function (Sequelize, DataTypes) {
...
-          if (instance.platformFeePercent === null || instance.platformFeePercent === undefined) {
-            // Automatically waive fees for collectives created under the COVID-19 category
-            if (instance.tags?.includes('covid-19')) {
-              instance.platformFeePercent = 0;
-            } else if (instance.type === 'USER' || instance.type === 'ORGANIZATION') {
-              instance.platformFeePercent = null;
-            } else {
-              instance.platformFeePercent = OC_FEE_PERCENT;
-            }
-          }
```
Comment (kewitz) [inline server/models/Collective.js:716]: Feels good to remove this exception. :relieved:

File: server/models/Collective.js [hunk 5]
```diff
@@ -1987,6 +1978,8 @@ export default function (Sequelize, DataTypes) {
     this.HostCollectiveId = null;
     this.isActive = false;
     this.approvedAt = null;
+    this.hostFeePercent = null;
+    this.platformFeePercent = null;
```
Comment (kewitz) [inline server/models/Collective.js:1982]: :+1:

**Review by kewitz — COMMENTED** [review]
Outside the wrong quotes in the migration, everything looks good. Any exceptions in the DB? (Like COVID collectives hosted in a host with hostFeePercent)?

File: migrations/20201020153732-update-default-feePercent.js (hunk 1 above)
Comment (znarf) [reply migrations/20201020153732-update-default-feePercent.js:44]: Good one!

**Comment by znarf** [comment]
> Any exceptions in the DB? (Like COVID collectives hosted in a host with hostFeePercent)?

We're not touching them for now but they might lose their specific hostFeePercent  when swapping host.

**Pull request merged**